src/parser/jobs/sch/modules/Gauge.tsx

```diff
@@ -0,0 +1,103 @@
+import {t} from '@lingui/macro'
+import {Trans} from '@lingui/react'
+import Color from 'color'
+import {DataLink} from 'components/ui/DbLink'
+import {JOBS} from 'data/JOBS'
+import {Event} from 'event'
+import {EventHook} from 'parser/core/Dispatcher'
+import {filter, oneOf} from 'parser/core/filter'
+import {dependency} from 'parser/core/Injectable'
+import {CounterGauge} from 'parser/core/modules/Gauge'
+import {Gauge as CoreGauge} from 'parser/core/modules/Gauge/Gauge'
+import Suggestions, {SEVERITY, TieredSuggestion} from 'parser/core/modules/Suggestions'
+import React from 'react'
+
+const GAUGE_GAIN_AMOUNT = 10
+// Graph color
+const GRAPH_COLOR = Color(JOBS.SCHOLAR.colour)
+// Start at 30 because you can overcap when seraph is out and you can't drain the gauge
+const OVERCAP_SEVERITY = {
+	30: SEVERITY.MINOR,
+	50: SEVERITY.MEDIUM,
+	70: SEVERITY.MAJOR,
+}
+export class FaerieGauge extends CoreGauge {
+	static override handle = 'fairieGauge'
+	static override title = t('sch.gauge.title')`Faerie Gauge Usage`
+
+	@dependency private suggestions!: Suggestions
+	// Defaults
+	private gauge = this.add(new CounterGauge({
+		chart: {label: 'Faerie Gauge', color: GRAPH_COLOR},
+	}))
+	private fairyOut: boolean = false
+	private actorPets = this.parser.pull.actors.filter(actor => actor.owner != null && actor.owner.id === this.parser.actor.id).map(pet => pet.id)
+	private petHook?: EventHook<{ type: 'action'; action: number; source: string; target: string; timestamp: number; }>
+
+	override initialise() {
+		super.initialise()
+		const playerFilter = filter<Event>().source(this.parser.actor.id)
+
+		//Faerie Summon
+		this.addEventHook(playerFilter.type('action').action(oneOf([this.data.actions.SUMMON_EOS.id, this.data.actions.SUMMON_SELENE.id])), this.onSummon)
+		//sanity check, your pets can't take actions if they're not out.
+		this.petHook = this.addEventHook(filter<Event>().source(oneOf(this.actorPets)), this.onSummon)
```

Comment:
This assumes they summoned pre-pull. Not sure that's safe unless you're determining pet backwards in some way. I'd also make a petFilter like your playerFilter.

Reply:
I specifically asked for it to be this way, because we only need to check if the pet exists pre-pull until we see it.  This is the best way to do it without an adapter (which isn't really necessary for SCH pets because they don't do anything that complicated.